### PR TITLE
 fix join_point bug

### PR DESCRIPTION
**Pull request by qinzuoyan**

### CODE REVIEW

File: src/core/core/address.cpp [hunk 1]
```diff
@@ -150,13 +150,13 @@ DSN_API const char*   dsn_address_to_string(dsn_address_t addr)
             );
         break;
     case HOST_TYPE_URI:
-        p = (char*)addr.u.uri.uri;
+        p = (char*)(long)addr.u.uri.uri;
```
Comment (imzhenyu) [inline src/core/core/address.cpp]: sizeof(long) == 4 on Windows, so better to use (uintptr_t) instead

File: src/core/core/address.cpp [hunk 2]
```diff
@@ -150,13 +150,13 @@ DSN_API const char*   dsn_address_to_string(dsn_address_t addr)
...
         break;
     case HOST_TYPE_GROUP:
-        p = (char*)(((dsn::rpc_group_address*)(addr.u.group.group))->name());
+        p = (char*)(((dsn::rpc_group_address*)(long)(addr.u.group.group))->name());
```
Comment (imzhenyu) [inline src/core/core/address.cpp]: same as above.

File: src/core/core/address.cpp (hunk 1 above)
Comment (qinzuoyan) [inline src/core/core/address.cpp]: done

File: src/core/core/address.cpp (hunk 2 above)
Comment (qinzuoyan) [inline src/core/core/address.cpp]: done

**Comment by imzhenyu** [comment]
very nice:)

**Pull request merged**